src/OrchardCore/OrchardCore.Abstractions/Shell/Removing/ShellRemovingContext.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+
+namespace OrchardCore.Environment.Shell.Removing;
+
+public class ShellRemovingContext
+{
+    private string _errorMessage;
+
+    public ShellSettings ShellSettings { get; set; }
+    public bool LocalResourcesOnly { get; set; }
+    public bool FailedOnLockTimeout { get; set; }
+    public bool Success => ErrorMessage == null;
+
+    public string ErrorMessage
```

Comment:
Since this is displayed to users, shouldn't this be localized?

Reply:
There will be a more global localized message added to the notifier.

Where this error message will be passed as a parameter to the localizer.

Hmm, more considered as technical info but I will look at it again.

Reply:
Here how it is used in the Tenants AdminController

    await _notifier.ErrorAsync(H["Failed to remove the tenant '{0}'. {1}",
        id,
        context.ErrorMessage]);

Let me know what you think

Reply:
Yep, I've seen that, and this is why I think this should be `LocalizedString` instead, since the message will be directly displayed to the user on the admin UI. Unlike an exception message that's just logged, this is not that kind of just technical info.

Reply:
Notice that this error message is computed to also include the exception message (not all the stack trace) as I thought it is useful to have more info here that usually we only log.

Hmm, no problem to use an `IStringLocalizer` in tenant level handler so that we can use `S["..."]` to set the error message, but `ShellRemovalManager` is a host level service as `ShellHost`, so there is not necessarily a registered `IStringLocalizer` that we could inject without throwing an exception.

Reply:
I see, indeed. Hmm, then at least have the option open for localization here with an `S[]` wrap, even if that's not a good translation authoring experience.

Reply:
Anyway interesting, I will try to find a solution. As I remember at the tenant level we register null localizer services so that it doesn't fail without the localization module, I think we just need to do the same at the host level, registering null localizer services in OCBuilder.ApplicationServices, which still allows people to register what they want in their app startup.

Reply:
Oops this is already the case, will see

Reply:
Okay done ;)